app/src/main/ui/src/hooks/useGroups.tsx

```diff
@@ -1,9 +1,9 @@
-import {GroupsService} from "../services/Groups";
-import { useAPI} from "./useAPI";
-import type {QueryOptions} from "./useAPI";
-import type {GroupsModel} from "../services/Groups";
+import { GroupsService} from "../services/Groups";
 
-export const useGroups = (opts: QueryOptions):{groups: GroupsModel, isLoading: boolean} => {
-    const {data, isLoading} = useAPI<GroupsModel, QueryOptions>(GroupsService.list, opts);
-    return {groups: data, isLoading: isLoading};
-};
+
+import {usePaginatedData} from "./usePaginatedData";
+import type {Group} from "../services/Groups";
+
+export default function useGroups() {
+    return usePaginatedData<Group[]>(GroupsService.list, {pageSize: 20, start: 0});
```

Comment:
How we can override the pageSize or start from the caller of the function? Should add these parameters to the hook itself?

Reply:
The usePaginatedData hook is simple enough to be called directly if the behaviour should be changed somewhere. The useGroups hook was created with the purpose to keep all logic outside of the components and is only used at only one location while i dont think this will change. So i dont think it is necessary or helpful to allow the override. What do you think?